File: audio/audio-Rcode/audio.R
  library(warbleR)
  global1 <- data.frame(matrix(ncol = 25, nrow = 0))
  colnames(global1)=c("class","sound.files","selec","duration","meanfreq","sd","median","Q25","Q75",
  "IQR","skew","kurt","sp.ent","sfm","mode",
  "centroid","peakf","meanfun","minfun","maxfun","meandom","mindom","maxdom","dfrange","modindx")
  
  ##########################################################################################
  
  setwd("/home/lawrence/udemy/audio/Audio/Hello/")
  
  for (i in 32:48){
  	print(i)
  	track  = paste("Track",i,sep="")
  	track  = paste(track,".wav",sep="")
  	print(getwd())
  	X = data.frame(sound.files=track,selec=0,start=0,end=1, bottom.freq=0, top.freq=44000)
  	a <- specan(X)
  	a["class"] = "Hello"
  	print(a)
  	global1 = rbind(global1,a)
  }

##########################################################################################

setwd("/home/lawrence/udemy/audio/Audio/Goodbye")

for (i in 54:66){
	print(i)
	track  = paste("Track",i,sep="")
	track  = paste(track,".wav",sep="")
	X = data.frame(sound.files=track,selec=0,start=0,end=1, bottom.freq=0, top.freq=44000)
	a <- specan(X)
	a["class"] = "Goodbye"
	print(a)
	global1 = rbind(global1,a)
}

##########################################################################################

setwd("/home/lawrence/udemy/audio/Audio/Banana")

for (i in 70:87){
	print(i)
	track  = paste("Track",i,sep="")
	track  = paste(track,".wav",sep="")
	X = data.frame(sound.files=track,selec=0,start=0,end=1, bottom.freq=0, top.freq=44000)
	a <- specan(X)
	a["class"] = "Banana"
	print(a)
	global1 = rbind(global1,a)
}

##########################################################################################

setwd("/home/lawrence/udemy/audio/Audio/Chair")

for (i in 106:118){
	print(i)
	track  = paste("Track",i,sep="")
	track  = paste(track,".wav",sep="")
	X = data.frame(sound.files=track,selec=0,start=0,end=1, bottom.freq=0, top.freq=44000)
	a <- specan(X)
	a["class"] = "Chair"
	print(a)
	global1 = rbind(global1,a)
}

##########################################################################################

setwd("/home/lawrence/udemy/audio/Audio/IceCream")

for (i in 89:101){
	print(i)
	track  = paste("Track",i,sep="")
	track  = paste(track,".wav",sep="")
	X = data.frame(sound.files=track,selec=0,start=0,end=1, bottom.freq=0, top.freq=44000)
	a <- specan(X)
	a["class"] = "IceCream"
	print(a)
	global1 = rbind(global1,a)
}

########################################################################################## 



global2 = na.omit(global1)

write.csv(global2,"/home/lawrence/udemy/audio/Audio/data_model.csv")


##########################################################################################
##########################################################################################

validation <- data.frame(matrix(ncol = 25, nrow = 0))
colnames(validation )=c("class","sound.files","selec","duration","meanfreq","sd","median","Q25","Q75",
"IQR","skew","kurt","sp.ent","sfm","mode",
"centroid","peakf","meanfun","minfun","maxfun","meandom","mindom","maxdom","dfrange","modindx")

setwd("/home/lawrence/udemy/audio/Audio/validation")
path="/home/lawrence/udemy/audio/Audio/validation"

file.names <- dir(path, pattern =".wav")
for(i in 1:length(file.names)){
  filename = unlist(strsplit(file.names[i],"_"))[1]
  X = data.frame(sound.files=file.names[i],selec=0,start=0,end=1, bottom.freq=0, top.freq=44000)
  a <- specan(X)
  a["class"] = filename
  validation = rbind(validation ,a)
}
	
###########################################################################################

validation = na.omit(validation)

write.csv(validation ,"/home/lawrence/udemy/audio/Audio/validation.csv")  

File: audio/audio_.py
import pandas as pd
import numpy as np
from sklearn import tree
from sklearn.tree import DecisionTreeClassifier, export_graphviz
from sklearn import preprocessing
from sklearn.ensemble import AdaBoostClassifier
from sklearn import cross_validation
from sklearn import linear_model
from sklearn.metrics import confusion_matrix
from sklearn.svm import SVC,LinearSVC,NuSVC


 
train    = pd.read_csv("C:\\Udemy\\Practical Data Science in Python\\Audio\\data_model.csv")
test     = pd.read_csv("C:\\Udemy\\Practical Data Science in Python\\Audio\\validation.csv")

#------------------------------------------------------------------------------------------------#

x = train[["meanfreq","sd","median","Q25","Q75","IQR","skew","kurt","sp.ent","sfm","mode","centroid",
           "peakf","meanfun","minfun",
       "maxfun","meandom","mindom","maxdom","dfrange","modindx"]]
y = train["class"]
clh = tree.DecisionTreeClassifier(max_depth  = 7)
clf = AdaBoostClassifier(base_estimator= clh,n_estimators=10)
clf.fit(x,y)

print(np.mean(cross_validation.cross_val_score(clf, x, y, cv=8)))
print(confusion_matrix(y, clf.predict(x)))
test_x = test[["meanfreq","sd","median","Q25","Q75","IQR","skew","kurt","sp.ent","sfm","mode",
               "centroid","peakf","meanfun","minfun",
       "maxfun","meandom","mindom","maxdom","dfrange","modindx"]]
print(clf.score(test_x,test["class"]))

#---------------------------------------------------------------------------#

svcfit = SVC(C=0.01, kernel='linear', gamma='scale')
#x     =  preprocessing.scale(x)

svcfit.fit(x, y)

print(np.mean(cross_validation.cross_val_score(svcfit, x, y, cv=8)))
print(confusion_matrix(y, svcfit.predict(x)))
#test_x     =  preprocessing.scale(test_x)
print(svcfit.score(test_x,test["class"]))

#---------------------------------------------------------------------------#




